packages/mwp-api-state/src/sync/index.js

```diff
@@ -64,85 +52,74 @@ export function getDeprecatedSuccessPayload(successes, errors) {
  * metadata about the navigation action can also be sent to the `apiRequest`
  * here.
  *
+ * note that this function executes _downstream_ of reducers, so the
+ * new `routing` data has already been populated in `state`
+ *
  * @param {Object} routes The application's React Router routes
  * @returns {Function} an Epic function that emits an API_REQUEST action
  */
 export const getNavEpic = (routes, baseUrl) => {
 	const resolveRoutes = getRouteResolver(routes, baseUrl);
-	return (action$, store) => {
-		return action$
-			.ofType(LOCATION_CHANGE, SERVER_RENDER)
-			.mergeMap(({ payload: location }) => {
-				// note that this function executes _downstream_ of reducers, so the
-				// new `routing` data has already been populated in `state`
-				const state = store.getState();
-				const { referrer = {} } = state.routing;
-				// inject request metadata from context, including `store.getState()`
-				const requestMetadata = {
-					referrer: referrer.pathname || state.config.entryPath || '',
-					logout: location.pathname.endsWith('logout'), // assume logout route ends with logout
-					clickTracking: state.clickTracking,
-					retainRefs: [],
-				};
-
-				const cacheAction$ = requestMetadata.logout
-					? Observable.of({ type: 'CACHE_CLEAR' })
-					: Observable.empty();
-
-				const resolvePrevQueries = referrer.pathname
-					? resolveRoutes(referrer, baseUrl).then(getMatchedQueries(referrer))
-					: Promise.resolve([]);
-				const resolveNewQueries = resolveRoutes(location, baseUrl).then(
-					getMatchedQueries(location)
-				);
-				const apiAction$ = Observable.fromPromise(
-					Promise.all([
-						resolveNewQueries,
-						resolvePrevQueries,
-					]).then(([newQueries, previousQueries]) => {
-						// perform a fast comparison of previous route's serialized queries
-						// with the new route's serialized queries. All state refs for
-						// _shared_ queries should be retained
-						const serializedNew = newQueries.map(JSON.stringify);
-						const serializedPrev = previousQueries.map(JSON.stringify);
-						const sharedRefs = serializedPrev
-							.filter(qJSON => serializedNew.includes(qJSON))
-							.map(JSON.parse)
-							.map(q => q.ref);
-						requestMetadata.retainRefs = sharedRefs;
-						return newQueries;
-					})
-				).map(q => api.requestAll(q, requestMetadata));
+	return (action, store) => {
+		if (![LOCATION_CHANGE, SERVER_RENDER].some(type => type === action.type)) {
+			return Promise.resolve([]);
+		}
+		const { payload: location } = action;
+		const state = store.getState();
+		const { referrer = {} } = state.routing;
+		// inject request metadata from context, including `store.getState()`
+		const requestMetadata = {
+			referrer: referrer.pathname || state.config.entryPath || '',
+			logout: location.pathname.endsWith('logout'), // assume logout route ends with logout
```

Comment:
would be nice to have more notes here about why/what this is for or a link to those notes

Reply:
okay - this is actually sort of legacy code when we were thinking about how the app should handle logout, but it hasn't been implemented in any application so I'll make that clear